File: gnome-sort/gnome-sort.py
def gnome_sort(array): #GNOME SORT
    i = 1
    while i < len(array): #Iterate until end of array
        if array[i] > array[i-1] or i == 0: #move "gnome" forward if "pots" in  correct order
            i += 1
        else:
            array[i], array[i-1] = array[i-1], array[i] #swap if needed, go back
            i -= 1

    return array


if __name__ == "__main__":
    import os, sys #import shuffler from parent directory
    dir_path = os.path.dirname(os.path.realpath(__file__))
    parent_dir_path = os.path.abspath(os.path.join(dir_path, os.pardir))
    sys.path.insert(0, parent_dir_path)
    from shuffler import random_sequence

    print("GNOME SORT")

    shuffled_array = random_sequence(0, 10)

    print(shuffled_array, "\n")

    print(gnome_sort(shuffled_array))

File: gnome-sort/gnome-sort.go
package main

import (
	"math/rand"
	"time"
	"fmt"
)

func random_sequence(minimum, maximum int) []int { //returns a shuffled array
	array := make([]int, 0)
	for i := minimum; i < maximum; i++ { //create array and append numbers to it
		array = append(array, i)
	}

	rand.Seed(time.Now().UnixNano())
	rand.Shuffle(len(array), func(i, j int) { array[i], array[j] = array[j], array[i] }) //shuffle the array

	return array
}

func gnome_sort(array []int) []int {
	i := 0
	for i < len(array) { //iterate until end of array
		if i == 0 || array[i] > array[i-1] { //move "gnome" forward if "pots" in  correct order
			i++
		} else {
			array[i], array[i-1] = array[i-1], array[i] //swap if needed, go back
            i--
		}

	}
	return array
}

func main() {
	shuffled_array := random_sequence(0, 1000)
	fmt.Println("GNOME SORT")
	fmt.Println(shuffled_array)
	fmt.Println()
	fmt.Println(gnome_sort(shuffled_array))
}